### PR TITLE
add Divio Aldryn Cloud service

### PR DESCRIPTION
**Pull request by stefanfoulis**
This adds **Divio Aldryn Cloud** as a service to [dnsimple](https://dnsimple.com/) and will be listed [here](https://dnsimple.com/domains/aldryncloud.com/applied_services)

It will end up looking roughly like this:

screen shot 2016-01-19 at 21 37 13

screen shot 2016-01-21 at 11 29 27

and the **Domain Settings** section on control.aldryn.com would be altered to have the additional **Aldryn Project Domain** section, which is always visible, even if there are no domains configured yet.

screen shot 2016-01-21 at 11 36 53

### CODE REVIEW

**Comment by weppos** [comment]
Hey @stefanfoulis, thanks for the patch! Since the service variables will soon be presented via API, and they are scoped under the service, I suggest to use a shorter name in place of `aldryn_project_domain`

See for example the [Heroku service](https://github.com/aetrion/dnsimple-services/blob/master/services/heroku/config.json#L19).

What about `domain`, or `appname`, or something similar?

**Comment by stefanfoulis** [comment]
Ok, I'll change it to a shorter name. My thinking for verbose naming was exactly to avoid possible naming conflicts.
:-)

**Comment by stefanfoulis** [comment]
@weppos wouldn't ``domain`` clash with the builtin var? from the README:

```
In addition to custom attributes, there are also the following variables:

domain - The domain name, such as example.com.
subdomain - The subdomain name, such as www.
```

**Comment by stefanfoulis** [comment]
@weppos oh, and I don't quite understand what ``default-subdomain`` is exactly. How does it work?

**Comment by weppos** [comment]
> @weppos wouldn't domain clash with the builtin var?

Good catch! What about `project`?

> @weppos oh, and I don't quite understand what default-subdomain is exactly. How does it work?

`default-subdomain` is used in case you want to force the service on a subdomain, and the user doesn't provide it.

Some services suggest to be enabled at a specific subdomain (as they are generally side-features). An example is a calendar application that may want to specify

```
{
  "config": {
    ...
    "default-subdomain": "calendar"
  },
```

This flag is kind of tricky because at the moment we don't properly handle the subdomain in our UI (hence services are generally attached to naked domain and the `default-subdomain` is somehow used to indicate which specific CNAME should be created).

In other words, in your case it could be blank (I see other services used "www").

**Comment by stefanfoulis** [comment]
@weppos thanks for the clarifications. I removed ``subdomain-required`` since from what you wrote it does not make sense for us. And I removed the ``aldryn_`` prefix from the variable name.

**Comment by stefanfoulis** [comment]
@weppos is there anything missing that I need to add to get merged? :-)

**Comment by stefanfoulis** [comment]
ping @weppos @aeden

**Comment by weppos** [comment]
Sorry for the silence @stefanfoulis, we're doing a little bit of work around the APIs and other components. We'll look into the PR soon. We try to schedule OSS contributions review periodically, like every couple of weeks.

**Comment by stefanfoulis** [comment]
good to know. Thanks @weppos :-)

File: services/aldryn/config.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      "name": "project_domain",
+      "label": "Aldryn Project Domain",
+      "description": "e.g mysite.us.aldryn.io",
+      "example": "mysite.us.aldryn.io"
+    }
+  ],
+  "records": [
+    {
+      "type": "ALIAS",
+      "content": "{{project_domain}}.aldryn.io",
```
Comment (jcaudle) [inline services/aldryn/config.json]: Because of the way you're prompting for the value (asking for the whole project domain and giving an example of `mysite.us.aldryn.io`), this should just be `"content": "{{project_domain}}",` The same goes for the `www` record. If you'd like to prompt for just the project name and zone so you can enforce that the record is always added for an `aldryn.io` URL, you should change the example above to be `mysite.us`.

**Comment by stefanfoulis** [comment]
thanks for the review @jcaudle . I fixed the issue.

**Comment by weppos** [comment]
@stefanfoulis I merged it. I made a small change, I replaced the ALIAS on the www with the CNAME.
https://github.com/aetrion/dnsimple-services/commit/da97886b14810e8c0b0ee58bdf5fa37eef078bff

Was there any reason to use an ALIAS? We generally tend to not add the `www` version (see Heroku), but in this case I left it as it was your submission. However, unless you are strictly required to use an ALIAS, the CNAME would be preferred here.
